kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/converter/gaitup/PhysilogUploadProcessorFactory.kt

```diff
@@ -44,17 +42,21 @@ open class PhysilogUploadProcessorFactory(
             val projectId = checkNotNull(record.data?.projectId) { "Project ID required to upload image files." }
             val userId = checkNotNull(record.data?.userId) { "Project ID required to upload image files." }
             val relativePath = Paths.get("$projectId/$userId/$TOPIC/${record.id}/$dateDirectory/$fileName")
-            val fullPath = rootPath.resolve(relativePath).normalize()
-            val url = advertisedUrl.resolve(fullPath.toString())
+            val fullPath = uploaderCreate().rootDirectory().resolve(relativePath).normalize()
+            val url = uploaderCreate().advertisedTargetUri().resolve(fullPath.toString())
 
             try {
-                uploaderCreate().upload(fullPath, inputStream)
-
+                if(uploaderCreate() is S3FileUploader) {
+                    uploaderCreate().upload(relativePath, inputStream, contents.size)
+                }
+                else {
+                    uploaderCreate().upload(fullPath, inputStream, contents.size)
+                }
```

Comment:
It would be good if all types could use the same value, either by passing the root path so the uploader can relativize/normalise the path itself or by passing both the fullPath and the relative path. In any case, running `uploaderCreate()` twice seems wasteful.